### PR TITLE
Update current user rights during migration

### PR DESCRIPTION
**Pull request by AdrienClairembault**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets |

### CODE REVIEW

File: src/Migration.php [hunk 1]
```diff
@@ -1241,6 +1241,17 @@ public function addRight($name, $rights = ALLSTANDARDRIGHT, $requiredrights = ['
             ),
             true
         );
+
+        // Update current user rights so he doesn't have to reload his session
```
Comment (cedric-anne) [inline src/Migration.php:1245]:
```suggestion
        // Update current user rights so he doesn't have to reload his session.
        // This ensure rights created by plugin install/update migrations are applied immediately.
```

File: src/Migration.php [hunk 2]
```diff
@@ -1241,6 +1241,17 @@ public function addRight($name, $rights = ALLSTANDARDRIGHT, $requiredrights = ['
...
+        if (!isCommandLine()) {
```
Comment (cedric-anne) [inline src/Migration.php:1246]:
```suggestion
        if (Session::getLoginUserID() !== false) {
```

**Review by cedric-anne — APPROVED** [review]

**Review by cedric-anne — CHANGES_REQUESTED** [review]
Is it possible to also update the `updateRight` and the `addRightByInterface` methods ?

**Comment by AdrienClairembault** [comment]
> Is it possible to also update the `updateRight` and the `addRightByInterface` methods ?

If the sessions has to be updated in multiple places, maybe it would be simpler to just reload the current session rights at the end of `Plugin->install()` ?

If I recall correctly, @flegastelois suggested a piece of code like `Session::changeProfile(curent_profile);` in our internal discussion which would do the trick.

**Comment by cedric-anne** [comment]
> > Is it possible to also update the `updateRight` and the `addRightByInterface` methods ?
> 
> If the sessions has to be updated in multiple places, maybe it would be simpler to just reload the current session rights at the end of `Plugin->install()` ?
> 
> If I recall correctly, @flegastelois suggested a piece of code like `Session::changeProfile(curent_profile);` in our internal discussion which would do the trick.

You could then probably declare a new `Session::reloadCurrentProfile()` method to do this, but it may have an unexpected impact on the currently active entity.

**Comment by AdrienClairembault** [comment]
Ok for this new method.

If we are unsure of the side effects then it should be moved to main as it is not a priority issue at all and can wait the 10.1 release.

**Comment by AdrienClairembault** [comment]
Regarding this comment from another PR:

> A simple solution could be to store the date of the last rights changes in the config table and in each user session.
Then if a user load a page and his "date of last rights updates" does not match the one in GLPI's config, we force reload his menu + rights.

This would allow "real time" updates for any already logged in users.
Would this solution be acceptable ?

**Comment by cedric-anne** [comment]
> Regarding this comment from another PR:
> 
> > A simple solution could be to store the date of the last rights changes in the config table and in each user session.
> > Then if a user load a page and his "date of last rights updates" does not match the one in GLPI's config, we force reload his menu + rights.
> 
> This would allow "real time" updates for any already logged in users. Would this solution be acceptable ?

Adding a date column on the glpi_profiles table would not cost much, and the check/refresh could probably be done in `Session::checkValidSessionId()` https://github.com/glpi-project/glpi/blob/7d5ad9eb12eb0508c3924bd2a2c4d403d43893e7/src/Session.php#L928

The current requests that checks if profile is still attach to user could probably be replaced by a request that check if profile is still attached and/or has change, and then reload rights if it is necessary.

**Comment by cedric-anne** [comment]
> Would this solution be acceptable ?

See my previous comment.

Anyway, it would not be acceptable on bugfixes branch.

**Comment by AdrienClairembault** [comment]
> Anyway, it would not be acceptable on bugfixes branch.

Of course.

I'll close this one then since it would no longer be relevant with this new suggested solution.

**Pull request closed without merging**